### PR TITLE
Join login

### PR DESCRIPTION
**Pull request by mirrou**
Initial Join and Login forms

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+    "extends": "airbnb",
+    "rules": {
+      "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
+      "linebreak-style": 0
```
Comment (reggieroby) [inline .eslintrc.js]: a few you might add:

"complexity": 1
"eqeqeq": 1,
"no-else-return": 1,
"no-unused-vars": 1,
"indent": [1, 2, {
      "VariableDeclarator": {
        "var": 2,
        "let": 2,
        "const": 3
      }
    }],
"newline-after-var": 1,
"no-mixed-spaces-and-tabs": 1,
"no-tabs": 1,
"prefer-rest-params": 1,
"prefer-spread": 1,

**Comment by reggieroby** [comment]
you should have started this branch off of master(3186e33) and some of this would not have been needed, such as the package.json here.

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react';
+import './App.css';
+import {
+  BrowserRouter as Router,
+  Route,
+  Link,
+} from 'react-router-dom'
+
+import Join from './components/Join';
```
Comment (reggieroby) [inline src/App.js]: you should use named imports as a rule. only use default imports when using a 3rd party library.

for ex: 
`import Join from './....' ` would be a default import.
`import {Join} from './....' ` would be a named import.

i will leave a comment about how to export it as a named export in the Join comp

File: src/components/Join.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+        <FormItem {...tailFormItemLayout}>
+          <Button type="primary" htmlType="submit">Join</Button>
+        </FormItem>
+      </Form>
+    );
+  }
+}
+
+const WrappedRegistrationForm = Form.create()(RegistrationForm);
```
Comment (reggieroby) [inline src/components/Join.js]: export const WrappedRegistrationForm = Form.create()(RegistrationForm);

that should work. in general export your variable declaration and its a named export.

File: src/components/Home.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    <p className="App-intro">
+            Get your travel on!
+    </p>
+  </div>
+);
+
+export default Home;
+
+
+// import { Form, Icon, Input, Button, Checkbox } from 'antd';
```
Comment (reggieroby) [inline src/components/Home.js]: you should copy this branch onto another local branch if you want to perserve the comments. and remove these comments from this branch before merging into master. you might never get back to this.

File: src/components/Join.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+      if (!err) {
+        console.log('Received values of form: ', values);
+      }
+    });
+  }
+  handleConfirmBlur = (e) => {
+    const value = e.target.value;
+    this.setState({ confirmDirty: this.state.confirmDirty || !!value });
+  }
+  checkPassword = (rule, value, callback) => {
```
Comment (reggieroby) [inline src/components/Join.js]: although your checkPassword and confim function look correct, i think you have them switched for which form field they should be used on. your checkconfirm function is being used on the password input validator, and checkPassword is being used on the confirm password input.

File: src/App.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import Login from './components/Login';
+import Home from './components/Home';
+
+const App = () => (
+<Router>
+  <div>
+    <ul>
+      <li><Link to="/">Home</Link></li>
+      <li><Link to="/join">Join</Link></li>
+      <li><Link to="/Login">Login</Link></li>
```
Comment (reggieroby) [inline src/App.js]: all your routes should be lowercase when possible, or atleast consistent among all your paths. both setup and your href links as well.

File: src/components/Login.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+            valuePropName: 'checked',
+            initialValue: true,
+          })(
+            <Checkbox>Remember me</Checkbox>
+          )}
+          <a className="login-form-forgot" href="">Forgot password</a>
+          <Button type="primary" htmlType="submit" className="login-form-button">
+            Login
+          </Button>
+          Or <a href="./Join">Join Now!</a>
```
Comment (reggieroby) [inline src/components/Login.js]: your routehandler specifies a lowercase `./join`

File: src/App.js (hunk 2 above)
Comment (jvaughn575) [reply src/App.js]: @reggieroby On the server side, I could not get the above import statements to work. An error was thrown and it said it did not recognize the 'import' key. So I used 'commonJS' imports with the require keyword instead. What configuration file do I need to change, so I can use the es6 import syntax instead?

File: .eslintrc.js (hunk 1 above)
Comment (jvaughn575) [reply .eslintrc.js]: Cool, I need some eslint help. My code feels a bit all over the place.

File: src/App.js (hunk 2 above)
Comment (reggieroby) [reply src/App.js]: you just need babel-node for development, so in package.json you might have: `NODE_ENV=development ./node_modules/nodemon/bin/nodemon.js --exec ./node_modules/babel-cli/bin/babel-node.js ./server.js`
along with a .babelrc file containing at minimum ```{
  "presets": ["env","stage-0"]
}
```
with the presets you'll install those as npm packages. this is all setup for you in create react app behind the scenes.

Comment (jvaughn575) [reply src/App.js]: @reggieroby According to this issue we should probably keep all files in the src directory. I tried the above but couldn't get it to work. https://github.com/facebookincubator/create-react-app/issues/1699

Comment (jvaughn575) [reply src/App.js]: @reggieroby Finally got it to work. I have npm transpile the server.js file and save it to server-transpiled.js. Then I run nodemon on the transpiled file. Installed babel and babel-cli as dev dependences, and added the .babelrc file with the presets you suggested to the same level as the server.js file.

Comment (reggieroby) [reply src/App.js]: the server files you are creating should not be in src. src is for the client side code used in react. this is a separate "project" that we are just using the same repo to store. it should be in its own folder, call it "server" at the root level with src folder.

Comment (reggieroby) [reply src/App.js]: if you transpile it, you lose the reload capability of nodemon. what your doing now is what you would want to do for a production build, probably something similar to this: 
```
./node_modules/.bin/babel src -d dist
```
but for development youll want to use the above so you dont have to rebuild every time you change a file.

File: .eslintrc.js (hunk 1 above)
Comment (reggieroby) [reply .eslintrc.js]: we will get there. this project has a lot of setup to get around too, but most important thing to focus on is delievering business value.

**Comment by jvaughn575** [comment]
Maroua created a new branch for this pull request, which has already been merged. As this one is no longer needed I am closing it.

**Pull request closed without merging**